src/fluree/db/db/json_ld.cljc

```diff
@@ -242,11 +243,12 @@
 
 (defn expand-iri
   "Expands an IRI from the db's context."
-  ([{:keys [schema] :as _db} iri]
-   (json-ld/expand-iri iri (:context schema)))
-  ([{:keys [schema] :as _db} iri provided-context]
-   (->> (json-ld/parse-context (:context schema) provided-context)
-        (json-ld/expand-iri iri))))
+  ([db iri]
+   (expand-iri db iri nil))
+  ([db iri provided-context]
+   (if (keyword? iri)
+     (json-ld/expand-iri iri (dbproto/-context db provided-context :keyword))
+     (json-ld/expand-iri iri (dbproto/-context db provided-context :string)))))
```

Comment:
I think this would simplify my work for https://github.com/fluree/http-api-gateway/issues/38. In order to support policy opts via http-api, I need to make sure that role iris will be correctly expanded/resolved to subject ids. That has required weaving in some `context-type` logic into places where it wasn't previously, but with this I think I wouldn't need to do that at all.